Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/UserDAO.java

```diff
@@ -389,4 +412,37 @@ public QuadrigaUserDeniedDTO getDeniedUser(String id) {
         return (QuadrigaUserDeniedDTO) sessionFactory.getCurrentSession().get(QuadrigaUserDeniedDTO.class, id);
     }
     
+    
+    public QuadrigaUserDTO findUserByProviderUserId(String userId, String provider) throws QuadrigaStorageException {
+        QuadrigaUserDTO userDTO  = null;
+        try {
+            Query query = sessionFactory.getCurrentSession().createQuery(
+                    " from QuadrigaUserDTO user where user.userIdOfProvider=:userId AND user.provider=:provider");
+            query.setParameter("userId", userId);
+            query.setParameter("provider", provider);
+            List resultList = query.list();
+            
+            if(resultList != null){
+                System.out.println("resultList.size() : "+resultList.size());
+                if(resultList.size() > 0)
+                    userDTO = (QuadrigaUserDTO)(resultList.get(0));
+                else 
+                    userDTO = null;
+            }
+            else
+                return null;
+            
+            return  userDTO;
+        } catch (Exception e) {
+            logger.error("Error in adding an account request: ", e);
```

Comment:
remove; the exception shoould be logged when handled. and is this really throwing exception?